src/arns.lua

```diff
@@ -355,28 +355,67 @@ function arns.getProcessIdForRecord(name)
 	return record ~= nil and record.processId or nil
 end
 
+local function getActiveArNSNamesBetweenTimestampsFromRecords(records, startTimestamp, endTimestamp)
+	local activeNames = {}
+	local namesInGracePeriod = {}
+
+	for name, record in pairs(records) do
+		if record.type == "permabuy" then
+			table.insert(activeNames, name)
+		elseif
+			record.type == "lease"
+			and record.endTimestamp
+			and record.startTimestamp
+			and record.startTimestamp <= startTimestamp
+		then
+			if record.endTimestamp >= endTimestamp then
+				table.insert(activeNames, name)
+			elseif record.endTimestamp + constants.gracePeriodMs >= endTimestamp then
```

Comment:
since we have `getActiveArNSNamesBetweenTimestamps`

consider making this `getGracePeriodArNSNamesBetweenTimestamps` and then change

`getTotalActiveAndGracePeriodArNSNamesBetweenTimestampsUnsafe `  - call both to get the totals

it's a little difficult to follow which functions are just wrappers vs doing the logic - so ideally make the call tree flatter (less nested calls to other functions)